### PR TITLE
24 admin method optimize

### PR DESCRIPTION
**Pull request by meox**
- retrieve_stats(as: :json)
- retrieve_stats_for(collection_name)
- optimize_collection(collection_name)
- optimize_collections(deleted_threshold = 10)

### CODE REVIEW

**Comment by meox** [comment]
- retrieve_stats(as: :json)
- retrieve_stats_for(collection_name) 
- optimize_collection(collection_name)
- optimize_collections(deleted_threshold = 10)

File: sunspot/lib/sunspot/admin_session.rb [hunk 1]
```diff
@@ -354,13 +354,131 @@ def add_failed_replica(collection:, shard:, node:)
...
+    end
+
+    #
+    # Retrieve stats for all collections
+    #
+    # @param [Symbol] :as <:json, :table>
+    #
+    # Example: retrieve_stats(as: :table)
+    #
+    def retrieve_stats(as: :json)
```
Comment (giovannelli) [inline sunspot/lib/sunspot/admin_session.rb:399]: In the cluster status report you use view:, what do you think to change view: in as: there?

File: sunspot/lib/sunspot/admin_session.rb [hunk 2]
```diff
@@ -354,13 +354,131 @@ def add_failed_replica(collection:, shard:, node:)
...
     end
 
+    #
+    # Retrieve stats for the given collection
+    #
+    # @param [String] collection_name: is the collection name
+    #
+    # @return [Hash] stats info
+    #
+    def retrieve_stats_for(collection_name)
```
Comment (giovannelli) [inline sunspot/lib/sunspot/admin_session.rb:364]: What about using the cache here? You could store the cache per collection with a TTL, after the optimisation you could expire and set the cache again.

File: sunspot/lib/sunspot/admin_session.rb (hunk 1 above)
Comment (meox) [reply sunspot/lib/sunspot/admin_session.rb:399]: done

**Comment by meox** [comment]
tested on staging

**Comment by meox** [comment]
see issue: #24

**Pull request merged**